t/native_cp_al/native_cp_al.t.in

```diff
@@ -0,0 +1,39 @@
+#!@PERL@
+use strict;
+use Test::More tests => 11; # pack here your amount of subtests
+use SysWrap;
+
+#
+# Test native_cp_al works with cmd_cp option not set and sync_first on (see contig).
+#
+
+ok(! remove_snapshot_root(),
+	" snapshot root does not exist before testing starts");
+
+ok(0 == rsnapshot("-c @TEST@/native_cp_al/conf/native_cp_al.conf sync 2>&1"),
+	" sync failed");
+ok(-d "@SNAP@/.sync/native_cp_al",
+	" sync failed");
+ok(0 == rsnapshot("-c @TEST@/native_cp_al/conf/native_cp_al.conf hourly 2>&1"),
+	" hourly backup after sync failed");
+ok(-d "@SNAP@/hourly.0/native_cp_al",
+	" hourly backup failed");
+
+#
+# Test native_cp_al works with cmd_cp option not set, sync_first on and link_dest on.
+#
+
+ok(! remove_snapshot_root(),
+	" snapshot root does not exist before testing starts");
+
+ok(0 == rsnapshot("-c @TEST@/native_cp_al/conf/native_cp_al_link_dest.conf sync 2>&1"),,
+	" sync failed");
+ok(-d "@SNAP@/.sync/native_cp_al",
+	" sync failed");
+ok(0 == rsnapshot("-c @TEST@/native_cp_al/conf/native_cp_al_link_dest.conf hourly 2>&1"),
```

Comment:
I would expect this to fail as well without your patch

Reply:
Nope. If `link_dest` is set the  `cp_al()` sub is not used in backup process and `cp_al()` is the only place that calls `rsync_cleanup_after_native_cp_al()` sub and only when `cp_al` is not set. On reflection I will just remove this test. It just causes confusion. Not sure why I added it in the first place.